quizzes/ch05-02-an-example-program-using-structs.toml

```
@@ -0,0 +1,38 @@
+[[questions]]
+type="Tracing"
+prompt.program = """
+#[derive(Drop)]
+struct Rectangle {
+    width: u64,
+    height: u64,
+}
+
+fn perimeter(r: Rectangle) -> u64 {
+    let perimeter = 2 * (r.width + r.height);
+    perimeter
+}
+
+fn main(){
+    let rectangle1 = Rectangle{
+        width: 20,
+        height: 10,
+    };
+    println!("{}", perimeter(rectangle1));
+}
+"""
+answer.doesCompile = true
+answer.stdout = "60"
```

it's interesting that you decided to not derive copy on Rectangle, so that `perimeter` takes ownership of it. I think it's more challenging and interesting to highlight this fact in the quiz

```suggestion
fn main(){
    let rectangle1 = Rectangle{
        width: 20,
        height: 10,
    };
    let p = perimeter(rectangle1);
    println!("2 * ({} + {}) = {}", rectangle1.width, rectangle1.height, p);
}
"""
answer.doesCompile = false
```


and that would indeed not compile!